crates/biome_js_analyze/src/semantic_analyzers/nursery/no_unused_imports.rs

```diff
@@ -0,0 +1,242 @@
+use crate::{
+    react::{is_global_react_import, ReactLibrary},
+    semantic_services::Semantic,
+    JsRuleAction,
+};
+use biome_analyze::{context::RuleContext, declare_rule, ActionCategory, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_semantic::ReferencesExtensions;
+use biome_js_syntax::{
+    binding_ext::AnyJsBindingDeclaration, AnyJsImportClause, JsFileSource, JsIdentifierBinding,
+    JsImport, JsImportNamedClause, JsLanguage, JsNamedImportSpecifierList, JsSyntaxNode, T,
+};
+use biome_rowan::{
+    AstNode, AstSeparatedList, BatchMutation, BatchMutationExt, NodeOrToken, SyntaxResult,
+};
+
+declare_rule! {
+    /// Disallow unused imports.
+    ///
+    /// Unused imports usually are the result of an incomplete refactoring.
+    /// The code fix can remove comments associated with an `import`.
+    /// See the last invalid example.
+    ///
+    /// There is one exception to the rule: the `React` import.
+    /// Importing the `React` variable was a mandatory pattern until some time ago:
+    /// For the time being this rule will ignore it,
+    /// but this **might change in the future releases**.
```

Comment:
Do you think we could just get rid of the React exception? Since when the issue is fixed in babel?

Reply:
> Do you think we could just get rid of the React exception?

Yes, we can add an option to the rule, an array of strings:

```json
{
	"noUnusedImports": {
		"level": "error",
		"options": {
			"allowedImports": [
				{ "bindings": ["React"], "import": "react" }
			]
		}
	}
}
```

So users can allow-list anything they want from any import. 

> Since when the issue is fixed in babel?

One of the recent versions. It's now possible writing React components without the `React` at the top of the file.

Reply:
> Yes, we can add an option to the rule, an array of strings

I mean without providing any options?

Have you other known use-cases for such a config?

Reply:
Only react as far as I know. 

Uhmmmm, I'm overthinking it. Let's keep the React exception.

Reply:
I think we can drop the `React` exception. If users report this issue, we can re-introduce the exception. Moreover, the rule is not recommended.

Reply:
I will merge this PR without the exception. If you think we really need this exception, I can open a follow-up PR.

Reply:
An option to ignore whatever we want would be preferable. I'm using Biome in the Roblox ecosystem (with Typescript), and we have a port of React called `Roact`. Of course, it keeps trying to remove this too.